### DISCUSSION TITLE
Multi thread read while another thread is writing

### DISCUSSION THREAD

**Discussion by OkobaPatino** [duckdb/duckdb#13453] [Q&A]
As far as I can read in this issue https://github.com/duckdb/duckdb/issues/77#issuecomment-1116756282, it should be possible to have multiple connections while they are on the same process.
Is that correct?
[Also here](https://duckdb.org/docs/connect/concurrency) it is said that 

> One process can both read and write to the database.

I like to write into the database and have another thread read. But for both memory and file databases, when I open the write connection, it will fail on the next connect for read only access.
And if I share the connection or database, prepare will fail with an exception.

**Comment by chu-k** [comment]
Are you perhaps creating multiple database instances, instead of connections? 

I was able to (in C++) pass the _same_ (read/write) Database instance to several threads which each create their own Connection. Similarly to this [Python example](https://duckdb.org/docs/guides/python/multiple_threads) from their docs.

**Reply by OkobaPatino** [reply]
I do not know about the Python flow, but using C api in Pascal, it throws an exception. I hoped there is a sample code in C or C++.

**Comment by robotslacker** [comment]
Maybe my tool can help you . 
https://github.com/robotslacker/slackerdb

It fake to be a pg database, but the real content behind it is actually duckdb. In this way, you can connect remotely and connect multiple processes.
